File: backend/src/main/java/com/farmstay/entity/Reservation.java
package com.farmstay.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("reservation")
public class Reservation {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String reservationNo;
    private Long userId;
    private Long farmhouseId;
    private Long packageId;
    private Long scheduleId;
    private LocalDate reserveDate;
    private Integer personCount;
    private String contactName;
    private String contactPhone;
    private String remark;
    private Integer status;
    private String cancelReason;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: frontend/src/api/reservation.ts
import request from '@/utils/request'

export function create(data: {
  farmhouseId: number
  packageId: number
  scheduleId: number
  reserveDate: string
  personCount: number
  contactName: string
  contactPhone: string
  remark?: string
}) {
  return request.post('/api/reservations', data)
}

export function getMyList(page: number, size: number, status?: number) {
  return request.get('/api/reservations', { params: { page, size, status } })
}

export function getDetail(id: number) {
  return request.get(`/api/reservations/${id}`)
}

export function cancel(id: number, reason: string) {
  return request.put(`/api/reservations/${id}/cancel`, { cancelReason: reason })
}
